src/bespokelabs/curator/request_processor/online_request_processor.py

```diff
@@ -0,0 +1,312 @@
+        completed_request_ids = set()
+        if os.path.exists(save_filepath) and resume:
+            with open(save_filepath, "r") as f:
+                for line in f:
+                    response = GenericResponse.model_validate_json(line)
+                    if not response.response_errors:
+                        completed_request_ids.add(response.generic_request.original_row_idx)
+
+        # Count total requests
+        total_requests = sum(1 for _ in open(generic_requests_filepath))
+
+        # Create progress bar
+        status_tracker.pbar = tqdm(
+            initial=len(completed_request_ids),
+            total=total_requests,
+            desc=f"Processing {self.__class__.__name__} requests",
+        )
+
+        # Use higher connector limit for better throughput
+        connector = aiohttp.TCPConnector(limit=10 * rpm)
+        async with aiohttp.ClientSession(
+            connector=connector
+        ) as session:  # Initialize ClientSession here
+            async with aiofiles.open(generic_requests_filepath) as file:
+                pending_requests = []
+
+                async for line in file:
+                    generic_request = GenericRequest.model_validate_json(line)
+
+                    if resume and generic_request.original_row_idx in completed_request_ids:
+                        status_tracker.num_tasks_already_completed += 1
+                        continue
+
+                    request = APIRequest(
+                        task_id=status_tracker.num_tasks_started,
+                        generic_request=generic_request,
+                        api_specific_request=self.create_api_specific_request(generic_request),
+                        attempts_left=max_attempts,
+                        prompt_formatter=self.prompt_formatter,
+                    )
+
+                    token_estimate = self.estimate_total_tokens(request.generic_request.messages)
+
+                    # Wait for capacity if needed
+                    while not status_tracker.has_capacity(token_estimate):
+                        await asyncio.sleep(0.1)
+
+                    # Consume capacity before making request
+                    status_tracker.consume_capacity(token_estimate)
+
+                    task = asyncio.create_task(
+                        self.process_single_request(
+                            request=request,
+                            session=session,
+                            retry_queue=queue_of_requests_to_retry,
+                            save_filepath=save_filepath,
+                            status_tracker=status_tracker,
+                        )
+                    )
+                    pending_requests.append(task)
+
+                    status_tracker.num_tasks_started += 1
+                    status_tracker.num_tasks_in_progress += 1
+
+                    # Process any retries that are in the queue
+                    while not queue_of_requests_to_retry.empty():
```

Comment:
this prioritizes retrying instead of moving to the next request in the file. 

We should probably have a better system of exponential back off for our retries?

Reply:
the old code https://github.com/bespokelabsai/curator/blob/main/src/bespokelabs/curator/request_processor/openai_online_request_processor.py#L313-L314

waited until the entire file of requests was exhausted before retrying. 

potentially an issue doing the new way because you retry a request right away 5 times rapidly

Reply:
Discussed offline, 

Pro of doing the old way is the cleaner logic.

Potential Con is that if the error is due to time consuming operations, the stragglers can only retry at the very end, the user may need to wait for a long time at the end, rather than retry during generation.

Reply:
retry logic back to the old version: https://github.com/bespokelabsai/curator/pull/141/commits/29ad907b69b4321ac56c6a643aaf5d094c5d7a9a\